src/briefcase/platforms/web/static.py

```diff
@@ -96,33 +116,333 @@ def _trim_file(self, path, sentinel):
-                    len(path.parts) > 1
-                    and path.parts[1] == "static"
-                    and path.suffix == ".css"
+                    len(parts) > 1
+                    and parts[1] == "static"
+                    and path.suffix.lower() == ".css"
                 ):
                     self.console.info(f"    Found {filename}")
-                    css_file.write(
-                        "\n/*******************************************************\n"
+
+                    # Show deprecation warning once per wheel
+                    if not legacy_css_warning:
+                        self.console.warning(
+                            f"    {wheelfile.name}: legacy '/static' CSS detected; "
+                            "treating as insert into briefcase.css; this legacy handling will be removed in the future."
+                        )
+                        legacy_css_warning = True
+
+                    try:
+                        css_text = wheel.read(filename).decode("utf-8")
+                    except UnicodeDecodeError as e:
+                        raise BriefcaseCommandError(
+                            f"{filename}: CSS content must be UTF-8 encoded"
+                        ) from e
+
+                    rel_inside = "/".join(path.parts[2:])
+                    contrib_key = f"{package_key} (legacy static CSS: {rel_inside})"
+
+                    # Add CSS content to briefcase.css insert slot
+                    target = "static/css/briefcase.css"
+                    insert = "CSS"
+                    pkg_map = inserts.setdefault(target, {}).setdefault(insert, {})
+                    if contrib_key in pkg_map and pkg_map[contrib_key]:
+                        pkg_map[contrib_key] += "\n" + css_text
+                    else:
+                        pkg_map[contrib_key] = css_text
+
+                # New deploy/inserts handling
+                elif len(parts) >= 3 and parts[1] == "deploy" and parts[2] == "inserts":
+                    self.console.info(f"    Found {filename}")
+                    rel_inside = "/".join(parts[3:])
+
+                    if not rel_inside or rel_inside.endswith("/"):
+                        self.console.debug(
+                            f"    {filename}: skipping, not a valid insert file."
+                        )
+                    else:
+                        if "~" not in rel_inside:
+                            self.console.debug(
+                                f"    {filename}: skipping, filename must match '<target>~<insert>'."
+                            )
+                        else:
+                            try:
+                                target, insert = rel_inside.split("~")
+                            except ValueError:
+                                self.console.debug(
+                                    f"    {filename}: skipping, filename must match '<target>~<insert>'."
+                                )
```

Comment:
These two checks are redundant - if `~` is in rel_inside, you *must* be able to split on `~`. 

If the concern is the case of having more than one `~` - you can pass in the number of splits to allow (`.split("~", 1)`, or `.rsplit("~", 1)` if you want to preserve the "last" ~ as the split character.